frontend/src/lib/api.ts

```diff
@@ -299,3 +299,266 @@ export function useAllSubmissions(limit: number = 20, status?: string) {
+  });
+}
+
+// Activity Types
+
+export const ActivityType = {
+  CONTENT_SUBMISSION: "CONTENT_SUBMISSION",
+  CONTENT_APPROVAL: "CONTENT_APPROVAL",
+  TOKEN_BUY: "TOKEN_BUY",
+  TOKEN_SELL: "TOKEN_SELL",
+  POINTS_REDEMPTION: "POINTS_REDEMPTION",
+  POINTS_AWARDED: "POINTS_AWARDED",
+} as const;
+
+export type ActivityType = (typeof ActivityType)[keyof typeof ActivityType];
+
+export interface UserActivityStats {
+  type: ActivityType;
+  feed_id: string | null;
+  user_id: number;
+  id: number;
+  data: any;
+  metadata: any | null;
+  createdAt: Date;
+  updatedAt: Date | null;
+  timestamp: Date;
+  submission_id: string | null;
+}
+
+export interface ListOfUserActivityStats {
+  activities: UserActivityStats[];
+}
+
+export interface ActivityQueryOptions {
+  timeRange?: string;
+  feedId?: string;
+}
+
+export interface CuratedFeed {
+  feed_id: string;
+  feed_name: string | null;
+  submissions_count: number;
+  curator_rank: number | null;
+  points: number;
+  data: unknown;
+  metadata: any | null;
+}
+
+export interface FeedRank {
+  curatorRank: number | null;
+  approverRank: number | null;
+}
+
+// Activity Hooks
+export interface AggregatedActivityStats {
+  totalSubmissions: number;
+  totalApprovals: number;
+  totalRejections: number;
+  approvalRate: number;
+}
+
+export function useMyActivity() {
+  const { web3auth } = useWeb3Auth();
+
+  return useQuery<AggregatedActivityStats>({
+    queryKey: ["my-activity"],
+    queryFn: async () => {
+      if (!web3auth) throw new Error("Web3Auth not initialized");
+      const authResult = await web3auth.authenticateUser();
+
+      const response = await fetch("/api/activity/user/me", {
+        headers: {
+          Authorization: `Bearer ${authResult.idToken}`,
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error("Failed to fetch user activity");
+      }
+
+      const data: ListOfUserActivityStats = await response.json();
+
+      // Calculate statistics from activities
+      const stats = data.activities.reduce(
+        (acc, activity) => {
+          if (activity.type === ActivityType.CONTENT_SUBMISSION) {
+            acc.totalSubmissions++;
+          } else if (activity.type === ActivityType.CONTENT_APPROVAL) {
+            acc.totalApprovals++;
+          }
+          return acc;
+        },
+        { totalSubmissions: 0, totalApprovals: 0 },
+      );
+
+      // Calculate rejections (submissions - approvals)
+      const totalRejections = stats.totalSubmissions - stats.totalApprovals;
+
+      // Calculate approval rate
+      const approvalRate =
+        stats.totalApprovals + totalRejections > 0
+          ? stats.totalApprovals / (stats.totalApprovals + totalRejections)
+          : 0;
+
+      return {
+        totalSubmissions: stats.totalSubmissions,
+        totalApprovals: stats.totalApprovals,
+        totalRejections,
+        approvalRate,
+      };
+    },
+    enabled: !!web3auth,
+  });
+}
+
+export function useUserActivity(
```

Comment:
This should be used by:

<img width="1309" alt="Screenshot 2025-05-15 at 9 07 42 AM" src="https://github.com/user-attachments/assets/ac401d55-a8f3-4688-a0e2-3448eb956b3e" />